File: selectionsort/go/selectionsort.go
package main

import "fmt"

func main() {
	arr := []int{1, 2, 3, 100, 5}
	fmt.Println(selectionsort(arr))
}
func findsmallest(arr []int) int {
	smallest := arr[0]
	smallest_index := 0
	for i := 1; i < len(arr); i++ {
		if arr[i] < smallest {
			smallest = arr[i]
			smallest_index = i
		}
	}
	return smallest_index
}
func selectionsort(arr []int) []int {
	NewArr := []int{}
	copiedArr := make([]int, len(arr))
	copy(copiedArr, arr)
	for i := 0; i < len(arr); i++ {
		smallest := findsmallest(copiedArr)
		NewArr = append(NewArr, copiedArr[smallest])
		copiedArr = append(copiedArr[:smallest], copiedArr[smallest+1:]...)

	}
	return NewArr
}

File: selectionsort/selectionsort.py
def findsmallest(arr):
    smallest = arr[0]
    smallest_index = 0 
    for i in range(1,len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i 
    return smallest_index
def selectionSort(arr):
    NewArr = []
    copiedArr = list(arr)
    for i in range(len(copiedArr)):
        smallest = findsmallest(copiedArr)
        NewArr.append(copiedArr.pop(smallest))
    return NewArr
arr = [1,2,100,9]
print(selectionSort(arr))
